client/extensions/woocommerce/app/order/order-payment/dialog.js

```diff
@@ -0,0 +1,226 @@
+import Notice from 'components/notice';
+import OrderRefundTable from './table';
+import PriceInput from 'woocommerce/components/price-input';
+import { sendRefund } from 'woocommerce/state/sites/orders/refunds/actions';
+
+class OrderPaymentCard extends Component {
+	static propTypes = {
+		isPaymentLoading: PropTypes.bool,
+		fetchPaymentMethods: PropTypes.func.isRequired,
+		isVisible: PropTypes.bool.isRequired,
+		order: PropTypes.shape( {
+			currency: PropTypes.string.isRequired,
+			id: PropTypes.number.isRequired,
+			refunds: PropTypes.array.isRequired,
+			status: PropTypes.string.isRequired,
+			total: PropTypes.string.isRequired,
+		} ),
+		paymentMethod: PropTypes.oneOfType( [ PropTypes.object, PropTypes.bool ] ),
+		sendRefund: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		toggleDialog: PropTypes.func,
+		translate: PropTypes.func.isRequired,
+	};
+
+	state = {
+		errorMessage: false,
+		refundTotal: 0,
+		refundNote: '',
+	};
+
+	componentDidMount = () => {
+		const { siteId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchPaymentMethods( siteId );
+		}
+	};
+
+	componentWillReceiveProps = newProps => {
+		const newSiteId = newProps.siteId;
+		const oldSiteId = this.props.siteId;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchPaymentMethods( newSiteId );
+		}
+	};
+
+	toggleDialog = () => {
+		this.setState( {
+			errorMessage: false,
+			refundTotal: 0,
+			refundNote: '',
+		} );
+		this.props.toggleDialog();
+	};
+
+	recalculateRefund = total => {
+		this.setState( { refundTotal: total } );
+	};
+
+	updateNote = event => {
+		this.setState( {
+			refundNote: event.target.value,
+		} );
+	};
+
+	sendRefund = () => {
+		const { order, paymentMethod, siteId, translate } = this.props;
+		const maxRefund = parseFloat( order.total ) + getOrderRefundTotal( order );
+		if ( this.state.refundTotal > maxRefund ) {
+			this.setState( {
+				errorMessage: translate( 'Refund must be less than or equal to the order total.' ),
+			} );
+			return;
+		} else if ( this.state.refundTotal <= 0 ) {
+			this.setState( { errorMessage: translate( 'Refund must be greater than zero.' ) } );
+			return;
+		}
+		this.toggleDialog();
+		const refundObj = {
+			amount: this.state.refundTotal + '', // API expects a string
+			reason: this.state.refundNote,
+			api_refund: paymentMethod && -1 !== paymentMethod.method_supports.indexOf( 'refunds' ),
+		};
+		this.props.sendRefund( siteId, order.id, refundObj );
+	};
+
+	renderCreditCard = () => {
+		const { isPaymentLoading, paymentMethod, translate } = this.props;
+		if ( isPaymentLoading ) {
+			return null;
+		}
+
+		if ( paymentMethod && -1 === paymentMethod.method_supports.indexOf( 'refunds' ) ) {
+			return (
+				<div className="order-payment__method">
+					<h3>{ translate( 'Manual Refund' ) }</h3>
+					<p>
+						{ translate(
+							"This payment method doesn't support automated refunds and must be submitted manually."
```

Comment:
Grammar nit. "...and must be submitted manually" is an object of the subject "this payment method". Please reword. You might need to just say something like "This payment method does not support automated refunds. After recording the refund here, you must remit the amount to the customer manually."